packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,101 @@ void main() {
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets('DropdownButtonFormFieldState internal InputDecorator isEmpty property behaviour is correct', (WidgetTester tester) async {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            decoration: InputDecoration(labelText: 'label', alignLabelWithHint: true),
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // isEmpty: true causes the label to be aligned with the input text
+    await tester.pumpAndSettle();
+    expect(tester.getTopLeft(find.text('label')).dy, 44.0);
+    expect(tester.getTopLeft(find.text('label')).dy, 44.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 60.0);
+
+    // [FormField] value is initially null.
+    expect(stateKey.currentState.value, equals(null));
+    // [InputDecorator] isEmpty property is true.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(true));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('label'));
+    await tester.pumpAndSettle();
+
+    // Tap outside the [DropdownButton] to close dropdown menu without selecting an option.
+    await tester.tapAt(Offset.zero);
+    await tester.pumpAndSettle();
+
+    // [FormField] value is still null.
+    expect(stateKey.currentState.value, equals(null));
+    // [InputDecorator] isEmpty property is still true.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(true));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('label'));
+    await tester.pumpAndSettle();
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+    // [InputDecorator] isEmpty property is now false.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(false));
```

Comment:
Instead of checking whether `isEmpty` is false, check the intended alignment here as well.